pkg/scheduler/backend/queue/scheduling_queue_test.go

```diff
@@ -1528,7 +1528,7 @@ func BenchmarkMoveAllToActiveOrBackoffQueue(b *testing.B) {
 						}
 					}
 
-					ctx, cancel := context.WithCancel(context.Background())
+					ctx, cancel := context.WithCancel(ctx)
 					defer cancel()
```

Comment:
🤔 this one is tricky. We should probably call cancel inside each iteration of `for i := 0; i < b.N; i++`

Currently, because `defer` is called when the function terminates, we are leaking a routine from one iteration to the next.

Maybe we'll have to keep using `defer` but wrap the rest of the code of the function inside another function.

```golang
for i := 0; i < b.N; i++ {
  // setup
  func() {
    ctx, cancel := context.WithCancel(ctx)
    defer cancel()
    // test body
  }()
```